src/application-status-overlay/ApplicationStatusOverlayProvider.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Scroll from 'terra-scroll';
+import StatusView from 'terra-status-view';
+import ApplicationStatusOverlayContext from './ApplicationStatusOverlayContext';
+import styles from './ApplicationStatusOverlayProvider.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The components to be rendered within the context of the ApplicationStatusOverlayProvider.
+   * Components rendered here are able to interact with ApplicationStatusOverlayProvider through
+   * the ApplicationStatusOverlayContext.
+   */
+  children: PropTypes.node,
+  /**
+   * A function to be called with the current ref of the scrollable element rendered within the
+   * ApplicationStatusOverlayProvider.
+   */
+  scrollRefCallback: PropTypes.func,
+};
+
+const ApplicationStatusOverlayProvider = ({ children, scrollRefCallback, ...customProps }) => {
+  const [registeredStatusOverlay, setRegisteredStatusOverlay] = React.useState({});
+
+  const refContainer = React.useRef();
+
+  const contextValue = useMemo(() => ({
+    show: (key, data) => {
+      setRegisteredStatusOverlay(state => (
+        {
+          ...state,
+          [`${key}`]: data,
+        }
+      ));
+    },
+    hide: (key) => {
+      setRegisteredStatusOverlay((state) => {
+        const newRegisteredStatusOverlay = { ...state };
+        delete newRegisteredStatusOverlay[key];
+
+        return newRegisteredStatusOverlay;
+      });
+    },
+  }), []);
+
+  const disableContainerChildrenFocus = () => {
+    if (refContainer.current && refContainer.current.querySelector('[data-status-overlay-container-content]')) {
```

Comment:
You could put the ref on the container element [here](https://github.com/cerner/terra-application/pull/47/files#diff-ac0fedc9c031cc8f126738a12803ff63R106) and lose the queries.

Reply:
Changed here : https://github.com/cerner/terra-application/pull/47/commits/5e92dd7f93fe585a4883c6aa4734cb2859e0fa06